core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -124,39 +133,21 @@ public static KeyStoreManager getInstance(int tenantId,
      * @return KeyStore object
      * @throws Exception If there is not a key store with the given name
      */
-    public KeyStore getKeyStore(String keyStoreName) throws Exception {
+    public KeyStore getKeyStore(String keyStoreName) {
 
-        if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
-            return getPrimaryKeyStore();
-        }
-
-        if (isCachedKeyStoreValid(keyStoreName)) {
-            return loadedKeyStores.get(keyStoreName).getKeyStore();
-        }
-
-        String path = RegistryResources.SecurityManagement.KEY_STORES + "/" + keyStoreName;
-        if (registry.resourceExists(path)) {
-            org.wso2.carbon.registry.api.Resource resource = registry.get(path);
-            byte[] bytes = (byte[]) resource.getContent();
-            KeyStore keyStore = KeyStore.getInstance(resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
-            CryptoUtil cryptoUtil = CryptoUtil.getDefaultCryptoUtil();
-            String encryptedPassword = resource
-                    .getProperty(RegistryResources.SecurityManagement.PROP_PASSWORD);
-            String password = new String(cryptoUtil.base64DecodeAndDecrypt(encryptedPassword));
-            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
-            keyStore.load(stream, password.toCharArray());
-            KeyStoreBean keyStoreBean = new KeyStoreBean(keyStore, resource.getLastModified());
-            resource.discard();
+        try {
+            if (KeyStoreUtil.isPrimaryStore(keyStoreName)) {
+                return getPrimaryKeyStore();
+            }
 
-            if (loadedKeyStores.containsKey(keyStoreName)) {
-                loadedKeyStores.replace(keyStoreName, keyStoreBean);
-            } else {
-                loadedKeyStores.put(keyStoreName, keyStoreBean);
+            if (KeyStoreUtil.isCustomKeyStore(keyStoreName)) {
+                return getCustomKeyStore(keyStoreName);
             }
-            return keyStore;
-        } else {
-            throw new SecurityException("Key Store with a name : " + keyStoreName + " does not exist.");
+
+            return getTenantKeyStore(keyStoreName);
+        } catch (Exception e) {
+            log.error("Error loading the key store : " + keyStoreName);
```

Comment:
No need to log the error here, since we should be printing the error message from the place where the exception is handled

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/c1687dc983cee4418f175db98e13c1744c08a510